### PR TITLE
Feature permission

### PR DESCRIPTION
**Pull request by evrardco**
Fix for #358,
This pull request adds the possibility to check that users have a certain role before using any command.
Leaving that role to "undefined" falls back to the default free for all behavior.

### CODE REVIEW

**Comment by Perdition0** [comment]
Can you add command setd?

**Comment by evrardco** [comment]
> 
> 
> Can you add command setdj?

Wouldn't it be equivalent to add the dj role to someone ? Server owners can natively do that. If not, can you specify what you mean?

**Comment by Perdition0** [comment]
What I mean is that each particular server can define an exclusive roleDj
Example:
 /setdj @test
This server becomes its own Role @test

**Comment by evrardco** [comment]
I think that would require a more complex implementation. Something like a persistent storage that could map roles to servers. It's far from impossible but too complex to simply add it quickly. Maybe for another pull request :)

**Review by wolf-yuan-6115 — CHANGES_REQUESTED** [review]
So we can create a role called undefined

**Comment by evrardco** [comment]
Haha Idk why someone would do it but you're right 👍

**Comment by wolf-yuan-6115** [comment]
> Haha Idk why someone would do that but you're right 👍 

I think you need to 
```json
{
    "...":"",
    "DJ_ROLE":undefined
}
```

**Comment by Deepjq** [comment]
Yes plz add DJ role command 
Set DJ role
Remove DJ role

File: util/EvobotUtil.js [hunk 1]
```diff
@@ -26,3 +26,4 @@ exports.MAX_PLAYLIST_SIZE = config ? config.MAX_PLAYLIST_SIZE : process.env.MAX_
 exports.PRUNING = config ? config.PRUNING : process.env.PRUNING;
 exports.STAY_TIME = config ? config.STAY_TIME : process.env.STAY_TIME;
 exports.DEFAULT_VOLUME = config ? config.DEFAULT_VOLUME: process.env.DEFAULT_VOLUME;
+exports.DJ_ROLE = config ? config.DJ_ROLE : process.env.ROLE
```
Comment (drewburr) [inline util/EvobotUtil.js:29]: We should change `process.env.ROLE` to `process.env.DJ_ROLE` so this config isn't confusing for our Docker users

**Comment by razeNFR** [comment]
I have a problem with your feature, just the role "undefined" is working, so i have to create a role named "undefined" on my server to be working, if i change the name the server wont start.. I tried to replace "undefined" by "DJ_ROLE" but its not working i have the role "DJ_ROLE" in my server but its not working...

**Comment by eritislami** [comment]
Closing as this was left behind a long time now, will consider for the future

**Pull request closed without merging**